### ISSUE TITLE
Auto Import not working (Typescript ) after latest update

### ISSUE THREAD

**Issue by Sheldonfrith** [microsoft/vscode#132299]
Issue Type: <b>Bug</b>

Autocomplete imports have not worked for me since latest update. Specifically the list of suggestions that pops up when I'm typing, which normally I press "Tab" to select and automatically add the import statement at the top of the file. The suggestions are still there, but it no longer adds the import statement at the top of the file.

VS Code version: Code 1.60.0 (e7d7e9a9348e6a8cc8c03f877d39cb72e5dfb1ff, 2021-09-01T10:41:52.311Z)
OS version: Windows_NT x64 10.0.19042
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD Ryzen 9 3900X 12-Core Processor             (24 x 3800)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|31.92GB (14.25GB free)|
|Process Argv|--crash-reporter-id 34049ddb-b0b3-47a2-b7b3-23771ba5709f|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (54)</summary>

Extension|Author (truncated)|Version
---|---|---
better-comments|aar|2.1.0
namespace|adr|1.1.2
to|ami|0.0.10
laravel-extra-intellisense|ami|0.6.1
vscode-django|bat|1.6.0
vscode-intelephense-client|bme|1.7.1
vscode-tailwindcss|bra|0.6.14
simple-react-snippets|bur|1.2.4
npm-intellisense|chr|1.4.0
path-intellisense|chr|2.4.0
vscode-svgviewer|css|2.0.0
vscode-eslint|dba|2.1.23
netcore-snippets|dog|3.15.1
python-extension-pack|don|1.6.0
vscode-html-css|ecm|1.10.2
vscode-npm-script|eg2|0.3.22
prettier-vscode|esb|8.1.0
comment-anchors|Exo|1.9.6
php-debug|fel|1.17.0
php-intellisense|fel|2.3.14
auto-rename-tag|for|0.1.8
vscode-pull-request-github|Git|0.30.0
gc-excelviewer|Gra|3.0.44
csharpextensions|jch|1.3.1
vscode-position|jtr|1.0.3
auto-comment-blocks|kev|1.0.1
vscode-github|Kni|0.30.6
json-trimmer|log|0.0.3
bash-ide-vscode|mad|1.11.0
MagicPython|mag|1.1.0
vscode-wrap-console-log|mid|1.7.2
vscode-docker|ms-|1.16.1
csharp|ms-|1.23.15
python|ms-|2021.9.1191016588
vscode-pylance|ms-|2021.9.0
jupyter|ms-|2021.8.2031190714
jupyter-keymap|ms-|1.0.0
remote-ssh|ms-|0.65.7
remote-ssh-edit|ms-|0.65.7
vscode-remote-extensionpack|ms-|0.21.0
laravel-blade|one|1.25.0
laravel5-snippets|one|1.13.0
material-icon-theme|PKi|4.10.0
phpdoc-comment-vscode-plugin|rex|1.3.0
lit-plugin|run|1.2.1
vscode-autohotkey|sle|0.2.2
addDocComments|ste|0.0.8
msbuild-project-tools|tin|0.3.15
highlight-matching-tag|vin|0.10.1
vscodeintellicode|Vis|1.2.14
jinja|who|0.0.8
change-case|wma|1.0.0
JavaScriptSnippets|xab|1.8.0
convert-object-to-jsx|zac|1.0.0

(2 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383:30185418
pythonvspyt602:30300191
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492:30256859
pythonvspyt639:30300192
pythontb:30283811
pythonvspyt551:30345470
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
pythondataviewer:30285071
pythonvsuse255:30340121
vscod805:30301674
pythonvspyt200:30340761
vscextlangct:30333562
binariesv615:30325510
vsccppwtct:30329789
pythonvssor306:30344512
bridge0708:30335490
pygetstartedt2:30360495
bridge0723:30353136
pythonrunftest32:30361774
pythonf5test824cf:30361778
vsdyn321cf:30356811
pydsgst2:30361792
vscus224:30358039
vssid140:30362080

```

</details>

**Comment by dariusj18** [comment]
Similar issue for me, except it's an issue where in my monorepo it is only suggesting imports using the package path even for files from the same package.

ex. If I have two packages `@qwerty/foo` and `@qwerty/bar` and I have a class `Process`  in a file `process.ts` in the foo package, when I import it into bar it appropriatly imports it as `import { Process } from '@qwerty/foo'` but if I'm in foo, it _should_ import it as `import { Process } from './relative/path/process.ts'` but it is doing the former.

**Comment by dariusj18** [comment]
To clarify, this is new bug for this version and did not exist in the prior release, I have downgraded for now.

**Comment by dariusj18** [comment]
@viczam yes, just downloaded the prior version and reinstalled

**Comment by mjbvz** [comment]
Please share a minimal example project that demonstrates the issue

**Comment by artu-ole** [comment]
@mjbvz Do you imply that it somehow functions in 1.60.0 for you?
For me and others in this ticket I believe this is the case:
1.60.0
![SBKINtmuS8](https://user-images.githubusercontent.com/15938416/132635863-0aeccc12-d781-4ce9-a902-0878740cc4a6.gif)
1.59.1
![IdLc7hH6mM](https://user-images.githubusercontent.com/15938416/132635883-49b5355c-662a-49cd-b632-bc172416f993.gif)

To clarify, Quick Fix import action functions on 1.60.0, it's autocomplete that does not.

**Comment by mjbvz** [comment]
Please share a minimal example project with steps to reproduce the issue so I can test

**Comment by artu-ole** [comment]
@mjbvz I hope this helps https://github.com/artu-ole/vscode-132299

**Comment by Sheldonfrith** [comment]
Sorry @mjbvz, I really don't have time to create an example project. But hopefully this additional information helps:

I'm using latest version of NextJS, with this as my tsconfig.json: 
```
{
  "compilerOptions": {
    "target": "es5",
    "lib": ["dom", "dom.iterable", "esnext"],
    "allowJs": true,
    "skipLibCheck": true,
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "noEmit": true,
    "esModuleInterop": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "jsx": "preserve",
    "baseUrl": "."
  },
  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx"],
  "exclude": ["node_modules"]
}
```

### This only seems to be an issue in .tsx (React JSX) files. Regular .ts files the import statement is automatically added to the top as expected.

**Comment by gportela85** [comment]
Also had to downgrade VS Code, in our project, we have a `main.ts` that does
`export { ClassFoo } from '../../somefolder/foo';`

Before, Auto Import used to also import from `somefolder/foo`, but now it's importing from `main.ts` all around my code.

**Comment by eglavin** [comment]
Same issue visible in image below
Vscode: 1.60 on top
Vscode: 1.59.1 on bottom

![Untitled](https://user-images.githubusercontent.com/29385958/133507032-ebb359d4-d037-43e0-a7b0-14af2b0f25b2.png)

**Comment by IHIutch** [comment]
Same here

1.60.1
<img width="494" alt="Screen Shot 2021-09-15 at 5 17 31 PM" src="https://user-images.githubusercontent.com/20825047/133510949-85823bab-0323-40c9-83d2-3d9e785e6c69.png">

1.59.1
<img width="785" alt="Screen Shot 2021-09-15 at 5 18 19 PM" src="https://user-images.githubusercontent.com/20825047/133510948-030c9d0c-ffa2-4520-8e4c-dfb1fd79e1a4.png">

**Comment by mjbvz** [comment]
@artu-ole  Works for me:

![recording](https://user-images.githubusercontent.com/12821956/133528265-baa2823b-9923-4549-ab83-e93ce1e2572b.gif)

Any extra steps? Have you tested on insiders with extensions disabled?

@IHIutch Please share a minimal example project that demonstrates the issue. Also confirm that whatever package you are trying to import from is listed in your package.json (previously we also included imports for dependencies  of dependencies, but TS 4.4 tried to cut these down)

**Comment by IHIutch** [comment]
Here you go: https://github.com/IHIutch/auto-import-test

I also made a recording:
https://user-images.githubusercontent.com/20825047/133536156-6f4e487c-9a36-4708-885f-fcd181fb7fbb.mp4

It does look like the auto import works when you're using the specific package, but in previous VSCode versions, auto importing out of `@chakra-ui/react` worked.

This is VSCode 1.59.1:
https://user-images.githubusercontent.com/20825047/133536439-4b9b0c95-febe-4442-bf0a-e7ee3cbc32a3.mp4

**Comment by yudyananda** [comment]
Based on  [chakra-ui/chakra-ui#4608](https://github.com/chakra-ui/chakra-ui/discussions/4608) this behaviour may not only in typescript but  also happens in regular javascript.

**Comment by lillallol** [comment]
for me disabling lit-plugin extension and reloading vscode re enabled auto imports

**Comment by artu-ole** [comment]
@lillallol 
Thank you!

@mjbvz
Indeed, it's the lit-plugin that was causing the bug for me. 
I did not anticipate that vscode update would render a plugin incompatible to the point that all auto imports allover the app would break. Since as already mentioned it functions with the same plugins in 1.59.0. 
Hence I did not test without plugins. Nevertheless, my apologies, this is not a bug in vscode per say, at least in my environment.
I suspect there might be other plugins causing the same bug for others as well.

**Comment by jjspace** [comment]
I just wanted to add and confirm that this behavior was also happening for me for normal `.js` files and does not seem to be exclusive to typescript files.

I can also confirm that disabling `lit-plugin` on version 1.60+ made auto import work again. However with people above mentioning/showing react and chakra projects I would suspect this is not limited to the `lit-plugin` extension

**Comment by IHIutch** [comment]
I'll confirm that I do not have that extension installed

**Comment by Artur-** [comment]
I used the excellent ”extension bisect” to determine that lit-plugin caused the issue for me. You can probably find your problematic extension the same way

**Comment by Artur-** [comment]
This is how I can reproduce it
```
git clone https://github.com/Artur-/vscode-import-problem
cd vscode-import-problem
code .
```

1. Open both `exporter.js` and `importer.js`
2. In `importer.js` auto complete the import for `exportedMethod`

With lit-plugin installed and enabled: No import is added
With lit-plugin disabled: The import is added as expected

**Comment by IHIutch** [comment]
No such luck for me. As I mentioned above, auto import isn't completely broken for me. It just isn't working in a specific circumstance. I'm trying to understand if this is unexpected or expected behavior given the recent VSCode update.

Since I neither maintain `@chakra-ui/react` nor have a good understanding of Typescript and its intricacies, unfortunately, I won't be able to offer much help with debugging in that regard. Though I am happy to offer what information I can provide.

**Comment by jjspace** [comment]
@IHIutch If you have the time it might be helpful if you can try the [extension bisect](https://code.visualstudio.com/blogs/2021/02/16/extension-bisect) process @Artur- mentioned. You might be able to find another extension that's breaking it for you and maybe that will help find the core problem quicer?

**Comment by mjbvz** [comment]
@Artur- @jjspace Please file this issue against the 

@IHIutch Please open a separate issue with a minimal repo so we can investigate separately?

@Sheldonfrith Can you check if lit-html was causing this? If not, please share a minimal example with steps to reproduce so we can investigate this

**Comment by jjspace** [comment]
@mjbvz @Artur- already created one in the `lit-plugin`'s parent repo https://github.com/runem/lit-analyzer/issues/200

**Comment by Artur-** [comment]
Is this possibly related to `getCompletionEntryDetails` having an additional parameter in TS 4.4 that is not present at least in 4.2? https://github.com/runem/lit-analyzer/blob/56489d04242b35d692b1db4d0f776f9ca0da19b9/packages/ts-lit-plugin/src/ts-lit-plugin/ts-lit-plugin.ts#L45-L59 seems to cause the issue in `lit-plugin`

**Comment by Sheldonfrith** [comment]
@mjbvz  Yes it looks like this was caused by lit plugin for me as well. Thank you all! 
I should have done a thorough test of all my extensions before starting this issue. I would say this issue can be closed now.

**Comment by mjbvz** [comment]
Ok, I'm going to close this issue as caused by lit plugin 

If you are still seeing issue with the plugin/extension disabled, please open a new issue with a minimal example project demonstrating the issue so that we can investigate

**Comment by lpbayliss** [comment]
> Same here
> 
> 1.60.1
> <img alt="Screen Shot 2021-09-15 at 5 17 31 PM" width="494" src="https://user-images.githubusercontent.com/20825047/133510949-85823bab-0323-40c9-83d2-3d9e785e6c69.png">
> 
> 1.59.1
> <img alt="Screen Shot 2021-09-15 at 5 18 19 PM" width="785" src="https://user-images.githubusercontent.com/20825047/133510948-030c9d0c-ffa2-4520-8e4c-dfb1fd79e1a4.png">
> 
> That said, this is an example with https://github.com/chakra-ui/chakra-ui, other packages are still auto importing in 1.60.1, like https://github.com/react-hook-form/react-hook-form

@IHIutch I ran into a very simmilar problem with `@chakra-ui/react` where imports weren't working on autocomplete. Using `ctrl + .` to import the component did seem to work fine though.

I ran through some tests and discovered that having anything imported from `@chakra-ui/react` in the file was causing the issue. It appears the components need to be only imported from their respective submodule and `@chakra-ui/react` shouldn't be in the file at all.  I had manually imported something from `@chakra-ui/react` out of habit and caused myself to lose some hair.

**Comment by 43081j** [comment]
@mjbvz lit-plugin hasn't changed in some time, so its pretty clear that whatever was published in 1.60.0 of vscode was capable of breaking extensions which worked in 1.59.x, right?

if we haven't changed our extension and now it has broken, it is vscode's change, not ours. even if it turns out its some behaviour we were relying on that vsc dropped, thats a breaking change nobody announced.

ill look into it but i don't think its sensible to dismiss the issue here. vscode caused this, the extension did not change.

**Comment by mjbvz** [comment]
@43081j  This was caused by change in the internal TypeScript apis that lit-plugin extends, not by a change in VS Code itself. Try looking to see what changed with those apis in TS 4.4

**Comment by 43081j** [comment]
ah awesome, thanks for pointing me in the right direction. will take a look 👍
